.clang-format

```diff
@@ -0,0 +1,16 @@
+---
+BasedOnStyle: WebKit
+AllowShortBlocksOnASingleLine: false
+AllowShortFunctionsOnASingleLine: false
+AllowShortIfStatementsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: All
```

Comment:
We want AlwaysBreakAfterReturnType too which requires a recent clang-format.  I think we should ship a binary version for uniformity: it is changing too fast and we want too-recent features to rely on the distro package version.

Reply:
Right, I guess there's no ideal solution (llvm.org having clang-format as a separate download :)).  I guess we can explore this separately from this PR.

Reply:
It seems like http://apt.llvm.org/ provides APT mirrors for debian and ubuntu, which provide clang-format as a separate package (on X86 at least).